crates/sui-indexer/src/apis/read_api.rs

```diff
@@ -464,6 +468,62 @@ where
             .get_checkpoints(cursor, limit, descending_order)
             .await;
     }
+
+    async fn query_objects(
+        &self,
+        query: SuiObjectResponseQuery,
+        cursor: Option<ObjectID>,
+        limit: Option<usize>,
+        descending_order: Option<bool>,
+        at_checkpoint: Option<CheckpointId>,
+    ) -> RpcResult<ObjectsPage> {
```

Comment:
can you wrap the impl. here into query_objects_internal like `get_checkpoint_internal` internal above?
this abstraction would leave it easier in case we want to support non-PG compatible impl.